Scripts/BuildEngine.ps1

```diff
@@ -24,151 +24,152 @@
-    $CMakeCacheVariableOverride += " -DBUILD_SANDBOX_PROJECTS=ON"
-    $CMakeCacheVariableOverride += " -DENTT_INCLUDE_HEADERS=ON"
-    
-    # SDL2 options
-    #    
-    if ($systemName -eq "Linux") {
-        $CMakeCacheVariableOverride += " -DSDL_STATIC=ON"
-        $CMakeCacheVariableOverride += " -DSDL_SHARED=OFF"
-    }
-    
-    # Spdlog options
-    #
-    $CMakeCacheVariableOverride += " -DSPDLOG_BUILD_SHARED=OFF"
-    $CMakeCacheVariableOverride += " -DBUILD_STATIC_LIBS=ON"
-    $CMakeCacheVariableOverride += " -DSPDLOG_FMT_EXTERNAL=ON"
-    $CMakeCacheVariableOverride += " -DSPDLOG_FMT_EXTERNAL_HO=OFF"
-    
-    # GLFW options
-    #
-    if ($systemName -ne "Linux") {
-        $CMakeCacheVariableOverride += " -DGLFW_BUILD_DOCS=OFF"
-        $CMakeCacheVariableOverride += " -DGLFW_BUILD_EXAMPLES=OFF"
-        $CMakeCacheVariableOverride += " -DGLFW_INSTALL=OFF"
-    }
+    Write-Host "Building $systemName $architecture $configuration"
 
-    # ASSIMP options
-    #
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_TESTS=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_INSTALL=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_SAMPLES=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_ASSIMP_TOOLS=OFF"
-    
-    if ($systemName -eq "Darwin") {
-        $CMakeCacheVariableOverride += " -DBUILD_FRAMEWORK=ON"
-    }
+    [string]$BuildDirectoryNameExtension = If ($isMultipleConfig) { "MultiConfig" } Else { $configuration }
+    [string]$BuildDirectoryName = "Result." + $systemName + "." + $architecture + "." + $BuildDirectoryNameExtension
+    [string]$buildDirectoryPath = [IO.Path]::Combine($RepoRoot, $BuildDirectoryName)
+    [string]$cMakeCacheVariableOverride = ""
+    [string]$cMakeGenerator = ""
 
-    # STDUUID options
-    #
-    $CMakeCacheVariableOverride += " -DUUID_BUILD_TESTS=OFF"
-    $CMakeCacheVariableOverride += " -DUUID_USING_CXX20_SPAN=ON"
+    # Create build directory
+    if (-Not (Test-Path $buildDirectoryPath)) {
+        $Null = New-Item -ItemType Directory -Path $BuildDirectoryPath -ErrorAction SilentlyContinue
+    } 
+
+    $cMakeOptions = " -DCMAKE_SYSTEM_NAME=$systemName", " -DCMAKE_BUILD_TYPE=$configuration"
+    $submoduleCMakeOptions = @{
+        'ENTT'= @("-DENTT_INCLUDE_HEADERS=ON")
+        'SDL2' = @("-DSDL_STATIC=ON", "-DSDL_SHARED=OFF");
+        'SPDLOG' = @("-DSPDLOG_BUILD_SHARED=OFF", "-DBUILD_STATIC_LIBS=ON", "-DSPDLOG_FMT_EXTERNAL=ON", "-DSPDLOG_FMT_EXTERNAL_HO=OFF");
+        'GLFW '= @("-DGLFW_BUILD_DOCS=OFF", "-DGLFW_BUILD_EXAMPLES=OFF", "-DGLFW_INSTALL=OFF");
+        'ASSIMP'=@("-DASSIMP_BUILD_TESTS=OFF", "-DASSIMP_INSTALL=OFF", "-DASSIMP_BUILD_SAMPLES=OFF", "-DASSIMP_BUILD_ASSIMP_TOOLS=OFF");
+        'STDUUID'=@("-DUUID_BUILD_TESTS=OFF", "-DUUID_USING_CXX20_SPAN=ON");
+        'FRAMEWORK'=@("-DBUILD_FRAMEWORK=ON");
+    }  
+
+    $cMakeCacheVariableOverride = $cMakeOptions -join ' ' 
+
+    # Building CMake arguments switch to System
```

Comment:
```suggestion
    # Define CMake Generator argument 
```

Reply:
It's many argurments, so "Define CMake Generator arguments" instead of "Define CMake Generator argument"

Reply:
We define the CMake generator, which is a single value,
the generator and override aren't the same things

Reply:
j'ai ré-déposé ta variable à sa place